Source/Scene/Expression.js

```diff
@@ -697,12 +687,32 @@ define([
         return String(this._left.evaluate(feature));
     };
 
+    Node.prototype._evaluateRegExp = function(feature) {
+        if (!defined(this._value)) {
+            return new RegExp();
+        }
+
+        var pattern = this._value.evaluate(feature);
+        var flags = '';
+        if (defined(this._left)) {
+            flags = this._left.evaluate(feature);
+        }
+
+        var result;
+        try {
+            result = new RegExp(pattern, flags);
```

Comment:
Given that this will be slow, can we also add the optimization now for when literals are passed in?  Perhaps it will also be good infrastructure for future compile-time optimizations.